Microsoft/WebTemplateStudio
# angular npm run test should pass

This PR is to address the tests suggestions from #681 

Task:https://microsoftgarage.visualstudio.com/Intern%20GitHub/_sprints/taskboard/WTS%20Team/Intern%20GitHub/Y19-VS/05
Task:https://microsoftgarage.visualstudio.com/Intern%20GitHub/_sprints/taskboard/WTS%20Team/Intern%20GitHub/Y19-VS/05

(Since the task "master detail page has error" causes three of the tests failing, so I put the fix for that error in this PR too)

**My fix for test files:**
In Angular component test files, if we used some other components or modules in that component, than we need to "declare" those components and "import" the modules in the test file too. 

**My fix for the error:**
This error happened because one variable in master detail page did not get initialized, thus I add the initialization.

**How to test:**
1. generare a angular project with 4 types of pages selected
2. do npm install
3. do npm run test, all tests should all pass ( I selected 4 pages, thus 17 tests)
![ng-test](https://user-images.githubusercontent.com/12874317/59717398-c1114080-91cc-11e9-8e28-5cbc58ae48a1.PNG)
